pageserver/src/tenant/storage_layer.rs

```diff
@@ -343,7 +343,8 @@ impl LayerAccessStats {
 /// All layers should implement a minimal `std::fmt::Debug` without tenant or
 /// timeline names, because those are known in the context of which the layers
 /// are used in (timeline).
-pub trait Layer: std::fmt::Debug + Send + Sync {
+#[async_trait::async_trait]
```

Comment:
Another place of possible overhead, can be worked around if needed (you can return join handle from `get_value_reconstruct_data`). Actionable if needed.

Reply:
I think the mid-term plan is to use more `async` inside the `get_value_reconstruct_data` impl / eventually get rid of `get_value_reconstruct_data_blocking`. So, would like to avoid leaking the detail of spawn_blocking JoinHandle to the callers.